controller_parameter_server/CMakeLists.txt

```diff
@@ -13,27 +13,23 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(rcutils REQUIRED)
-find_package(yaml-cpp REQUIRED)
+find_package(yaml_cpp_vendor REQUIRED)
```

Comment:
you have to declare that dependency in the `package.xml` as well. That should solve the build failure.